homeassistant/components/purpleair/config_flow.py

```diff
@@ -184,6 +191,34 @@ async def async_validate_coordinates(
     return ValidationResult(data=nearby_sensor_results)
 
 
+async def async_validate_sensor_index(
+    hass: HomeAssistant,
+    api_key: str,
+    sensor_index: int,
+) -> ValidationResult:
+    """Validate a sensor index."""
+    api = async_get_api(hass, api_key)
+    errors = {}
+
+    try:
+        sensor_response = await api.sensors.async_get_sensor(
+            sensor_index, fields=["name"]
+        )
+    except NotFoundError:
+        errors["base"] = "unknown_sensor_index"
+    except PurpleAirError as err:
+        LOGGER.error("PurpleAir error while getting sensor: %s", err)
```

Comment:
Does these give any added value?

Reply:
Not sure I understand the question. Are you speaking about this log message? If so, yes: it logs what the error was, which is useful during debugging.

Reply:
Should it not be `LOGGER.debug` instead then?
Not sure the value of writing to log for a regular user as they already get feedback in the form

Reply:
This (writing errors to log) is a standard practice in config entries—the entry will only show the translated string, whereas the log entry can show additional info. It is not debug-specific.